docs/shared/recipes/nx-release/configure-custom-registries.md

```diff
@@ -0,0 +1,109 @@
+//my-default-registry.com/:_authToken=<DEFAULT_TOKEN>
+```
+
+With the above `.npmrc`, the following packages would be published to the specified registries:
+
+- `@my-scope/pkg-1` -> `https://my-custom-registry.com/`
+- `@other-scope/pkg-2` -> `https://my-other-registry.com/`
+- `pkg-3` -> `https://my-default-registry.com/`
+
+# Specify an Alternate Registry for a Single Package
+
+In some cases, you may want to configure the registry on a per-package basis instead of by scope. This can be done by setting options in the project's configuration.
+
+{% callout type="info" title="Authentication" %}
+All registries set for specific packages must still have authentication tokens set in the root `.npmrc` file for publishing in CI. See [Authenticate to the Registry in CI](#authenticate-to-the-registry-in-ci) for an example.
+{% /callout %}
+
+## Set the Registry in the Project Configuration
+
+The project configuration for Nx Release is in two parts - one for the version step and one for the publish step.
+
+### Update the Version Step
+
+The version step of Nx Release is responsible for determining the new version of the package. If you have set the `version.generatorOptions.currentVersionResolver` to 'registry', then Nx Release will check the remote registry for the current version of the package.
+
+**Note:** If you do not use the 'registry' current version resolver, then this step is not needed.
+
+To set custom registry options for the current version lookup, add the registry and/or tag to the `currentVersionResolverMetadata` in the project configuration:
+
+```json project.json
+{
+  "name": "pkg-5",
+  "sourceRoot": "...",
+  "targets": {
+    ...
+  },
+  "release": {
+    "version": {
+      "generatorOptions": {
+        "currentVersionResolverMetadata": {
+          "registry": "https://my-unique-registry.com/",
+          "tag": "next"
+        }
+      }
+    }
+  }
+}
+```
+
+### Update the Publish Step
+
+The publish step of Nx Release is responsible for publishing the package to the registry. To set custom registry options for publishing, you can add the `registry` and/or `tag` options for the `nx-release-publish` target in the project configuration:
+
+```json project.json
+{
+  "name": "pkg-5",
+  "sourceRoot": "...",
+  "targets": {
+    ...,
+    "nx-release-publish": {
+      "options": {
+        "registry": "https://my-unique-registry.com/",
+        "tag": "next"
+      }
+    }
+  }
+}
+```
+
+## Set the Registry in the Package Manifest
+
+{% callout type="caution" title="Caution" %}
+It is not recommended to set the registry for a package in the 'publishConfig' property of its 'package.json' file. 'npm publish' will always prefer the registry from the 'publishConfig' over the '--registry' argument. Because of this, the '--registry' CLI and programmatic API options of Nx Release will no longer be able to override the registry for purposes such publishing locally for end to end testing.
```

Comment:
```suggestion
It is not recommended to set the registry for a package in the 'publishConfig' property of its 'package.json' file. 'npm publish' will always prefer the registry from the 'publishConfig' over the '--registry' argument. Because of this, the '--registry' CLI and programmatic API options of Nx Release will no longer be able to override the registry for purposes such as publishing locally for end to end testing.
```